app/Policies/AlbumPolicy.php

```diff
@@ -172,7 +172,7 @@ public function canDownload(?User $user, ?BaseAlbum $baseAlbum): bool
 	 */
 	public function canEdit(User $user, ?AbstractAlbum $album): bool
 	{
-		if (!$this->userPolicy->canUpload($user)) {
+		if (!$this->userPolicy->mayUpload($user)) {
```

Comment:
Here we are inside a policy which decision for a right should be based on capabilities, security attributes and permissions, but not on rights from other policies.

```suggestion
		if (!$user->may_upload) {
```

Moreover you cannot upload a user, i.e. `UserPolicy::mayUpload` should simply not exist at all, because the user policy should returns rights on (other) user objects, not capabilities of that user.

The same holds for all the lines below in this class.